include/rapidjson/prettywriter.h

```diff
@@ -184,8 +186,16 @@ class PrettyWriter : public Writer<OutputStream, SourceEncoding, TargetEncoding,
     //@{
 
     //! Simpler but slower overload.
-    bool String(const Ch* str) { return String(str, internal::StrLen(str)); }
-    bool Key(const Ch* str) { return Key(str, internal::StrLen(str)); }
+    template <typename T>
+    bool String(const T* const& str, RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return String(str, internal::StrLen(str)); }
+    template <typename T>
+    bool Key(const T* const& str, RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return Key(str, internal::StrLen(str)); }
+    
+    //! The compiler can give us the length of quoted strings for free.
+    template <typename T, size_t N>
+    bool String(const T (&str)[N], RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return String(str, N-1); }
```

Comment:
> In what way does it work unexpectedly?
In a way that it truncates one element from the array.

> that's invalid code invoking undefined behavior
As I already pointed out, it's not true. With your patch this code doesn't invoke UB (or elaborate how it does) and it's not invalid in any way (or elaborate how it is).
If you want users to assume that there's only an overload for c-strings and keep this new overload an implementation detail then it's not going to work because this new overload for arrays will be part of the public API and will allow users to write code as above.